src/arrays/ladderops.jl

```diff
@@ -22,10 +22,60 @@
     before_eye(lens, pivot) = speye(prod(lens[[1:pivot-1]]))
     after_eye(lens, pivot) = speye(prod(lens[[pivot+1:length(lens)]]))
     function eye_sandwich(lens, pivot, op)
-        return kron(before_eye(lens, pivot), 
-                    op, 
+        return kron(before_eye(lens, pivot),
+                    op,
                     after_eye(lens, pivot))
     end
 
+function positionop(n::Int)
+    cop = raiseop(n)
+    return scale!(1/sqrt(2.),cop+cop')
+end
+
+function momentumop(n::Int)
+    cop = raiseop(n)
+    return scale(im/sqrt(2.), cop-cop')
+end
+
+function spin_h1(j::Float64)
+    if (isinteger(2*j) && j>0)
+        m = reverse([-j:j-1])
+        N = length(m)+1
+        m_coeffs = [sqrt(j*(j+1.0) - (x+1.0)*x) for x in m]
+        return spdiagm(m_coeffs,1,N,N)
+    else
+        return error("j must be non-negative integer or half integer")
+  end
+end
+
+function spin_h2(j::Float64)
+    if (isinteger(2*j) && j>0)
+        m = reverse([-j:j])
+        N = length(m)
+        return spdiagm(m,0,N,N)
+    else
+        return error("j must be non-negative integer or half integer")
+  end
+end
+
+spin_jx(j::Float64) = QuArray(0.5*(spin_h1(j)+spin_h1(j)'))
+spin_jy(j::Float64) = QuArray(-0.5*im*(spin_h1(j)-spin_h1(j)'))
+spin_jz(j::Float64) = QuArray(spin_h2(j))
+spin_jp(j::Float64) = QuArray(spin_h1(j))
+spin_jm(j::Float64) = QuArray(spin_h1(j)')
+sigmax = 2.0 * spin_jx(0.5)
```

Comment:
Maybe `sigma_x` etc for consistency? (We could even use Unicode sigma in addition). And you should declare them const.